pages/api/talks.spec.js

```diff
@@ -0,0 +1,27 @@
+import { advanceTo, clear } from 'jest-date-mock';
+import { getMonth } from 'date-fns';
+
+import { getNextEmptyMonth } from './talks';
+
+describe('talks', () => {
+    beforeEach(() => {
+        advanceTo(new Date(2019, 12, 20));
+    });
+
+    describe('getNextEmptyMonth', () => {
+        it.each`
+            talks                                                                       | expected
+            ${[{ date: '2019-11-20' }]}                                                 | ${1}
+            ${[{ date: '2019-12-20' }, { date: '2020-01-20' }]}                         | ${2}
+            ${[{ date: '2019-12-20' }, { date: '2020-01-20' }, { date: '2020-02-20' }]} | ${3}
+            ${[{ date: '2019-12-20' }, { date: '2020-02-20' }]}                         | ${1}
+            ${[{ date: '2020-02-20' }]}                                                 | ${1}
+        `('should return the right month: $expected', ({ talks, expected }) => {

```

Comment:
What the hell is that? :D

Reply:
[it.each doc](https://jestjs.io/docs/en/api#describeeachtablename-fn-timeout)

Really power full function. When you write same tests but only params change you can use it.each. 
Jest expose two style :
- Tagged template literal, like above
- Array like 

```
it.each([
  [1, 1, 2],
  [1, 2, 3],
  [2, 1, 3],
])('.add(%i, %i)', (a, b, expected) => {
    expect(a + b).toBe(expected);
  });
```

Reply:
Not a big fan of the template as a data structure thing, but okay